base-template/contributors-guide.md

```diff
@@ -1,51 +1,250 @@
+   ```
+
+When you have the code downloaded, you need to set the upstream repository.
+This is important so that you can keep your local fork updated.
+
+#### Setting the upstream repository
+
+1. In your terminal, navigate to the directory where you checked out the code:
+
+   ```bash
+   > cd ~/workspace/incubator
+   ```
+
+1. Check out the `master` branch:
+
+   ```bash
+   > git checkout master
+   ```
+
+1. List the current remote branches:
+
+   ```bash
+   > git remote -v
+   ```
+
+   This command should list two remotes, both marked `origin`, like this:
+
+   ```bash
+   origin  https://github.com/<your_username>/incubator.git (fetch)
+   origin  https://github.com/<your_username>/incubator.git (push)
+   ```
+
+1. Add the incubator repo as an upstream:
+
+   ```bash
+   > git remote add upstream https://github.com/thegooddocsproject/incubator
+   ```
+
+1. Check:
+
+   ```bash
+   > git remote -v
+   ```
+
+   This command should now have the same two `origin` remotes as before, plus two more labelled `upstream`, like this:
+
+   ```bash
+   origin  https://github.com/<your_username>/incubator.git (fetch)
+   origin  https://github.com/<your_username>/incubator.git (push)
+   upstream  https://github.com/thegooddocsproject/incubator (fetch)
+   upstream  https://github.com/thegooddocsproject/incubator (push)
+   ```
+
+Now that you have your repository set up, you can create a branch to work on.
+
+#### Creating a branch
+
+1. In your terminal, navigate to the directory where you checked out the code:
+
+   ```bash
+   > cd ~/workspace/incubator
+   ```
+
+1. Fetch the branches in the upstream repository:
+
+   ```bash
+   > git fetch upstream
+   ```
+
+1. Check out your fork's `master` branch:
+
+   ```bash
+   > git checkout master
+   ```
+
+   This will show a message like this:
+
+   ```bash
+   Switched to branch `master`
+   ```
+
+1. Merge the changes from the upstream `master` branch, into your fork's `master` branch:
+
+   ```bash
+   > git merge upstream/master
+   ```
+
+   This shows some output that indicates what changes have merged in to your fork.
+1. Create a new branch for the work you want to do.
+   Make sure you give it an appropriate name, and include your username:
+
+   ```bash
+   > git checkout -b update-readme-username
+   ```
+
+Make your changes.
+When you have finished, save your changes, and create a pull request.
+
+Each branch and pull request must contain a single piece of work.
+For example, if you want to update a particular page and fix some typos across a lot of different pages, create a branch for the page update, make the changes, and do a pull request.
+Then create a new branch for the typos, and do a second pull request.
+This makes it easier for you to track your own work, and also for reviewers looking at your pull requests.
+
+#### Creating a pull request (PR)
+
+1. Save your changes in your text editor.
+1. In your terminal, add the changes to your commit:
+
+   ```bash
+   > git add .
```

Comment:
I'd suggest `git add README.md` the `git add .` could result in someone opening PRS with a bunch of other files they have floating in their environment.